File: leetcode/1078-remove-outermost-parentheses/solution.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        stack<char> stack;
        string result="";

        for(int i=0;i<s.size();i++){
            if(s[i]=='('){
                if(stack.empty()){
                    stack.push('(');
                    continue;
                }
                else{
                    stack.push('(');
                    result+='(';
                }
            }
            else{
                stack.pop();
                if(stack.empty()){
                    continue;
                }
                result+=')';
            }
        }
        return result;
    }
};

File: leetcode/1078-remove-outermost-parentheses/solution.py
class Solution(object):
    def removeOuterParentheses(self, s):
        string=""
        stack=[]
        for i in s:
            if (i=="("):
                if (stack==[]):
                    stack.append(i)
                    continue
                else:
                    string=string+i
                    stack.append(i)
            else:
                stack.pop()
                if(len(stack)==0):
                    continue
                string=string+i
                    
        return string

        """
        :type s: str
        :rtype: str
        """
        
